lib/service/servicecfg/app.go

```diff
@@ -95,6 +95,12 @@ type App struct {
 	// CORS defines the Cross-Origin Resource Sharing configuration for the app,
 	// controlling how resources are shared across different origins.
 	CORS *CORS
+
+	// Ports is a list of ports and port ranges that an app agent can forward connections to.
+	// Only applicable to TCP App Access.
+	// If this field is not empty, URI is expected to contain no port number and start with the tcp
+	// protocol.
+	Ports []PortRange
```

Comment:
I forgot to adjust [the callsite which converts `lib/service/servicecfg.App` into `api/types.AppV3`](https://github.com/gravitational/teleport/blob/a801fc8a10573a7ac2145edf462ebde5e798a960/lib/service/service.go#L5935-L5949) and add tests for it, as well as tests for YAML configuration, I'll do that soon.